packages/block-library/src/cover/index.php

```diff
@@ -52,15 +45,40 @@ function render_block_core_cover( $attributes, $content ) {
 			return $content;
 		}
 
-		$processor = new WP_HTML_Tag_Processor( $content );
+		$processor = new WP_HTML_Tag_Processor( '<div></div>' );
 		$processor->next_tag();
 
-		$styles         = $processor->get_attribute( 'style' );
-		$merged_styles  = ! empty( $styles ) ? $styles . ';' : '';
-		$merged_styles .= 'background-image:url(' . esc_url( $current_featured_image ) . ');';
+		$current_caption = get_the_post_thumbnail_caption();
+		if ( $current_caption ) {
+			$processor->set_attribute( 'role', 'img' );
+			$processor->set_attribute( 'aria-label', $current_caption );
+		}
+
+		$processor->add_class( 'wp-block-cover__image-background' );
+		if ( $attributes['hasParallax'] ) {
+			$processor->add_class( 'has-parallax' );
+		}
+		if ( $attributes['isRepeated'] ) {
+			$processor->add_class( 'is-repeated' );
+		}
+
+		$styles = 'background-position:' . isset( $object_position ) ? $object_position : '50% 50%' . ';';
```

Comment:
The default `50% 50%` comes from the default value returned from [`mediaPosition`](https://github.com/WordPress/gutenberg/blob/502a2d5bd25cc06af4241ad872c3bcef0d56fed5/packages/block-library/src/cover/shared.js#L27) in [`save`](https://github.com/WordPress/gutenberg/blob/502a2d5bd25cc06af4241ad872c3bcef0d56fed5/packages/block-library/src/cover/save.js#L81).

This default value isn't used for `object-position` in the JS, but it probably should be. However, that can't be changed without adding a deprecation for the cover block which I don't want to do as part of this bug fix.